### PR TITLE
feat: Add docsRepo

### PR DESCRIPTION
**Pull request by wangyi7099**
In some cases, project and doc are not in the same repo like Vue. So, and a `docsRepo` option  in config file to distinguish project and doc address.

### CODE REVIEW

**Comment by wangyi7099** [comment]
Did not see the const definition jut now

File: docs/default-theme-config/README.md [hunk 1]
```diff
@@ -226,6 +226,8 @@ module.exports = {
   themeConfig: {
     // Assumes GitHub. Can also be a full GitLab url.
     repo: 'vuejs/vuepress',
+    // Your docs repo address
```
Comment (ulivz) [inline docs/default-theme-config/README.md]: Optional, your docs repo address, fallback to repo when it's not given.

File: lib/default-theme/Page.vue [hunk 2]
```diff
@@ -57,18 +57,19 @@ export default {
...
       if (endingSlashRE.test(path)) {
         path += 'README.md'
       } else {
         path += '.md'
       }
 
-      if (repo && editLinks) {
-        const base = outboundRE.test(repo)
-          ? repo
-          : `https://github.com/${repo}`
```
[the remark this answers is not in the document]
Comment (abdullah) [reply lib/default-theme/Page.vue]: It can take a full URL to handle repositories

[the remark this answers is not in the document]
Comment (wangyi7099) [reply lib/default-theme/Page.vue]: yes it was preserved, I just changed the var name from  repo to docsRepo.

File: lib/default-theme/Page.vue [hunk 3]
```diff
@@ -57,18 +57,19 @@ export default {
         docsDir = '',
         docsBranch = 'master'
       } = this.$site.themeConfig
```
Comment (egoist) [inline lib/default-theme/Page.vue]: Add `docsRepo = repo` here seems clearer 😄

File: docs/default-theme-config/README.md [hunk 4]
```diff
@@ -226,6 +226,8 @@ module.exports = {
   themeConfig: {
     // Assumes GitHub. Can also be a full GitLab url.
     repo: 'vuejs/vuepress',
+    // Optional, your docs repo address, fallback to repo when it's not given.
```
Comment (egoist) [inline docs/default-theme-config/README.md]: <code>Defaults to &#x60;themeConfig.repo&#x60;</code>

File: lib/default-theme/Page.vue (hunk 3 above)
Comment (wangyi7099) [reply lib/default-theme/Page.vue]: Thanks

File: docs/default-theme-config/README.md (hunk 4 above)
Comment (wangyi7099) [reply docs/default-theme-config/README.md]: @egoist 
Seems the same meaning

Comment (ycmjason) [reply docs/default-theme-config/README.md]: It is clearer if u specify `themeConfig.repo`

**Pull request merged**